src/components/common/Expenditures/ExpenditureDetailsPage/ExpenditureClaimButton/ExpenditureClaimButton.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import { ExpenditureStatus } from '~gql';
+import { ActionTypes } from '~redux';
+import { ActionButton } from '~shared/Button';
+import { Colony, Expenditure } from '~types';
+
+interface ExpenditureClaimButtonProps {
+  colony: Colony;
+  expenditure: Expenditure;
+}
+
+const ExpenditureClaimButton = ({
+  colony,
+  expenditure,
+}: ExpenditureClaimButtonProps) => {
+  if (expenditure.status !== ExpenditureStatus.Finalized) {
+    return null;
+  }
+
+  const isFullyClaimed = !!expenditure.balances?.every(
+    (balance) => balance.amount === '0',
+  );
+
+  if (isFullyClaimed) {
+    return <div>There are no more payouts to claim</div>;
+  }
+
+  let nextClaimableAt: number | null = null;
+  const claimableSlots = expenditure.slots.filter((slot) => {
+    const claimableFrom = new Date(
+      ((expenditure.finalizedAt ?? 0) + (slot.claimDelay ?? 0)) * 1000,
+    ).getTime();
+
+    if (Date.now() >= claimableFrom) {
```

Comment:
I see what you mean about testing this being tricky. I suppose the solution is to only advance block time by a few seconds, so system time can catch up. I guess it's not a problem at all in production since block time can't be meddled with.

Reply:
We can get the actual block time like we do in the `fetchMotionTimeoutPeriods` lambda but it's an async call and I suppose an anti-pattern for us. And there shouldn't be a difference between block time and date.now() in production, right?

Reply:
I wanna say it shouldn't make a difference (and the worst case is that the transaction will fail during a small time window) but we'll only know once it's deployed to a real chain